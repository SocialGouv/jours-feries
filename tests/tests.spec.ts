import joursFeries from "../src/index";

test("should match 2018 results", () =>
  expect(joursFeries(2018)).toMatchSnapshot());

test("should match 2020 results", () =>
  expect(joursFeries(2020)).toMatchSnapshot());

test("should match 2018 alsace-moselle results", () =>
  expect(joursFeries(2018, { alsace: true })).toMatchSnapshot());

test("should match 2020 alsace-moselle  results", () =>
  expect(joursFeries(2020, { alsace: true })).toMatchSnapshot());

test("Lundi de Pâques 1954", () => {
  expect(joursFeries(1954)["Lundi de Pâques"]).toMatchSnapshot();
});

test("Lundi de Pâques 1981", () => {
  expect(joursFeries(1981)["Lundi de Pâques"]).toMatchSnapshot();
});

test("Lundi de Pâques 2049", () => {
  expect(joursFeries(2049)["Lundi de Pâques"]).toMatchSnapshot();
});
