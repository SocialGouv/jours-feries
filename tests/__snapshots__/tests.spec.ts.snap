// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lundi de Pâques 1954 1`] = `1954-04-14T23:00:00.000Z`;

exports[`Lundi de Pâques 1981 1`] = `1981-04-15T22:00:00.000Z`;

exports[`Lundi de Pâques 2049 1`] = `2049-04-14T22:00:00.000Z`;

exports[`should match 2018 alsace-moselle results 1`] = `
Object {
  "Armistice": 2018-11-10T23:00:00.000Z,
  "Ascension": 2018-05-08T22:00:00.000Z,
  "Assomption": 2018-08-14T22:00:00.000Z,
  "Fête Nationale": 2018-07-13T22:00:00.000Z,
  "Fête du travail": 2018-04-30T22:00:00.000Z,
  "Jour de l'an": 2017-12-31T23:00:00.000Z,
  "Lundi de Pentecôte": 2018-05-19T22:00:00.000Z,
  "Lundi de Pâques": 2018-03-31T22:00:00.000Z,
  "Noël": 2018-12-24T23:00:00.000Z,
  "Saint Étienne": 2018-12-25T23:00:00.000Z,
  "Toussaint": 2018-10-31T23:00:00.000Z,
  "Vendredi Saint": 2018-03-28T22:00:00.000Z,
  "Victoire des alliés": 2018-05-07T22:00:00.000Z,
}
`;

exports[`should match 2018 results 1`] = `
Object {
  "Armistice": 2018-11-10T23:00:00.000Z,
  "Ascension": 2018-05-08T22:00:00.000Z,
  "Assomption": 2018-08-14T22:00:00.000Z,
  "Fête Nationale": 2018-07-13T22:00:00.000Z,
  "Fête du travail": 2018-04-30T22:00:00.000Z,
  "Jour de l'an": 2017-12-31T23:00:00.000Z,
  "Lundi de Pentecôte": 2018-05-19T22:00:00.000Z,
  "Lundi de Pâques": 2018-03-31T22:00:00.000Z,
  "Noël": 2018-12-24T23:00:00.000Z,
  "Toussaint": 2018-10-31T23:00:00.000Z,
  "Victoire des alliés": 2018-05-07T22:00:00.000Z,
}
`;

exports[`should match 2020 alsace-moselle  results 1`] = `
Object {
  "Armistice": 2020-11-10T23:00:00.000Z,
  "Ascension": 2020-05-15T22:00:00.000Z,
  "Assomption": 2020-08-14T22:00:00.000Z,
  "Fête Nationale": 2020-07-13T22:00:00.000Z,
  "Fête du travail": 2020-04-30T22:00:00.000Z,
  "Jour de l'an": 2019-12-31T23:00:00.000Z,
  "Lundi de Pentecôte": 2020-05-26T22:00:00.000Z,
  "Lundi de Pâques": 2020-04-07T22:00:00.000Z,
  "Noël": 2020-12-24T23:00:00.000Z,
  "Saint Étienne": 2020-12-25T23:00:00.000Z,
  "Toussaint": 2020-10-31T23:00:00.000Z,
  "Vendredi Saint": 2020-04-04T22:00:00.000Z,
  "Victoire des alliés": 2020-05-07T22:00:00.000Z,
}
`;

exports[`should match 2020 results 1`] = `
Object {
  "Armistice": 2020-11-10T23:00:00.000Z,
  "Ascension": 2020-05-15T22:00:00.000Z,
  "Assomption": 2020-08-14T22:00:00.000Z,
  "Fête Nationale": 2020-07-13T22:00:00.000Z,
  "Fête du travail": 2020-04-30T22:00:00.000Z,
  "Jour de l'an": 2019-12-31T23:00:00.000Z,
  "Lundi de Pentecôte": 2020-05-26T22:00:00.000Z,
  "Lundi de Pâques": 2020-04-07T22:00:00.000Z,
  "Noël": 2020-12-24T23:00:00.000Z,
  "Toussaint": 2020-10-31T23:00:00.000Z,
  "Victoire des alliés": 2020-05-07T22:00:00.000Z,
}
`;
